src/AttachVS/AttachVs.cs

```diff
@@ -215,61 +223,51 @@ private static bool IsVsOrNull(Process process)
         return isVs;
     }
 
-    private static bool IsCorrectParent(Process currentProcess, Process parent)
+    private static Process? GetParentProcess(Process process)
     {
-        try
+        int id = GetProcessId(process);
+        if (id != -1)
         {
-            // Parent needs to start before the child, otherwise it might be a different process 
-            // that is just reusing the same PID.
-            if (parent.StartTime <= currentProcess.StartTime)
+            try
             {
-                return true;
-            }
-
-            Trace($"Process {parent.ProcessName} ({parent.Id}) is not a valid parent because it started after the current process.");
-            return false;
-        }
-        catch
-        {
-            // Access denied or process exited while we were holding the Process object.
-            return false;
-        }
-    }
-
-    private static Process GetParentProcess(Process process)
-    {
-        int id;
-        try
-        {
-            var handle = process.Handle;
-            var res = NtQueryInformationProcess(handle, 0, out var pbi, Marshal.SizeOf<PROCESS_BASIC_INFORMATION>(), out int size);
+                Process parent = Process.GetProcessById(id);
 
-            var p = res != 0 ? -1 : pbi.InheritedFromUniqueProcessId.ToInt32();
+                // Parent needs to start before the child, otherwise it might be a different process
+                // that is just reusing the same PID.
+                if (parent.StartTime <= process.StartTime)
+                {
+                    return parent;
+                }
 
-            id = p;
-        }
-        catch
-        {
-            id = -1;
+                Trace($"Process {parent.ProcessName} ({parent.Id}) is not a valid parent because it started after the current process.");
+            }
+            catch
+            {
+                // throws when parent no longer runs
```

Comment:
This catch is now not as narrow scoped as before, it can throw for more reasons I think.

Reply:
All the `IsCorrect` was inside a global try/catch and the call to `GetProcessById` was also inside a global try/catch so I have simply merged the 2.